Test suite fails with libarchive >= 3.7.5

With libarchive 3.7.4 the test suite passes, but upgrading past 3.7.5 fails:

```
def test_check_archiveentry_using_python_testtar():
>       check_entries(join(data_dir, 'testtar.tar'))

tests/test_entry.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_entry.py:95: in check_entries
    actual = list(get_entries(test_file))
tests/__init__.py:47: in get_entries
    for entry in arch:
libarchive/read.py:27: in __iter__
    r = read_next_header2(archive_p, entry._entry_p)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

retcode = -10, func = <_FuncPtr object at 0xffffa1c58590>
args = (187650195747216, 187650196019632)

    def check_int(retcode, func, args):
        if retcode >= 0:
            return retcode
        elif retcode == ARCHIVE_WARN:
            logger.warning(_error_string(args[0]))
            return retcode
        else:
>           raise archive_error(args[0], retcode)
E           libarchive.exception.ArchiveError: Damaged tar archive (errno=22, retcode=-10, archive_p=187650195747216)

libarchive/ffi.py:98: ArchiveError
FAIL: tests/test_entry.py:test_check_archiveentry_using_python_testtar
```

--- a/tests/test_entry.py
+++ b/tests/test_entry.py
@@ -9,7 +9,7 @@ import unicodedata
 
 import pytest
 
-from libarchive import memory_reader, memory_writer
+from libarchive import ArchiveError, memory_reader, memory_writer
 from libarchive.entry import ArchiveEntry, ConsumedArchiveEntry, PassedArchiveEntry
 
 from . import data_dir, get_entries, get_tarinfos
@@ -64,7 +64,13 @@ def test_check_ArchiveEntry_against_TarInfo():
 
 
 def test_check_archiveentry_using_python_testtar():
-    check_entries(join(data_dir, 'testtar.tar'))
+    # This test behaves differently depending on the libarchive version:
+    # 3.5, 3.6 and presumably all future versions reject the archive as damaged,
+    # whereas older versions accepted it.
+    try:
+        check_entries(join(data_dir, 'testtar.tar'))
+    except ArchiveError as e:
+        assert e.msg == "Damaged tar archive"
 
 
 def test_check_archiveentry_with_unicode_and_binary_entries_tar():

